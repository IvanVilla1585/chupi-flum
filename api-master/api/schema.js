// @flow
import { merge } from 'lodash';
import { makeExecutableSchema } from 'graphql-tools';
// import { withFilter } from 'graphql-subscriptions';
// import { pubsub, CREATED_EVENT_TOPIC } from './subscriptions';

//import schemas and resolvers
import userSchema from './users/schema';
import userResolvers from './users/resolvers';
import categorySchema from './categories/schema';
import categoryResolvers from './categories/resolvers';
import unitSchema from './units/schema';
import unitResolvers from './units/resolvers';
import presentationSchema from './presentations/schema';
import presentationResolvers from './presentations/resolvers';
import machineSchema from './machines/schema';
import machineResolvers from './machines/resolvers';
import processSchema from './process/schema';
import processResolvers from './process/resolvers';
import materialSchema from './materials/schema';
import materialResolvers from './materials/resolvers';
import providerSchema from './providers/schema';
import providerResolvers from './providers/resolvers';
import productSchema from './products/schema';
import productResolvers from './products/resolvers';

const rootSchema = [`
  type Query {
    # Return user logged, if nobody is logged return null
    me: User,
  }
  type Mutation {
    test (
      sw: String
    ) : String
  }

  schema {
    query: Query,
    mutation: Mutation
  }
`];

const rootResolvers = {
  Mutation: {
    test(_, params) {
      return params.sw || 'Hello';
    },
  },
};

// Put schema together into one array of schema strings
// and one map of resolvers, like makeExecutableSchema expects
const schema = [
  ...rootSchema, ...userSchema, ...categorySchema, ...unitSchema, ...presentationSchema, ...machineSchema, ...processSchema, ...materialSchema,
  ...providerSchema, ...productSchema
];
const resolvers = merge(
  rootResolvers, userResolvers, categoryResolvers, unitResolvers, presentationResolvers, machineResolvers, processResolvers, materialResolvers,
  providerResolvers, productResolvers
);

const executableSchema = makeExecutableSchema({
  typeDefs: schema,
  resolvers,
});

export default executableSchema;
