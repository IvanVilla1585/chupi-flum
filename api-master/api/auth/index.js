'use stict'
const jwt = require('jsonwebtoken');
const path = require('path');
const mongoose = require('mongoose');
const Promise = require('bluebird');
const passport = require('passport');
const passportJWT = require('passport-jwt');

const db = require('./lib/db');
const config = require('./config');

mongoose.Promise = Promise;

class SetupAuth {
  constructor(app){
    this.app = app;
    //this.sendMail = new Mail(config.sendGridApiKey);
    this.configureApp();
  }

  configureApp() {
    this.configureStrategy();
    this.configureRoutes();
  }

  async connectionBD(){
    const options = {useMongoClient: true};
    const connection = await mongoose.createConnection(config.db, options);

    // Expose user model
    return connection.model('User');
  }

  async configureStrategy(){
    let connection = await this.connectionBD();
    let ExtractJwt = passportJWT.ExtractJwt;
    let Strategy = passportJWT.Strategy;
    let params = {
      secretOrKey: config.secretToken,
      jwtFromRequest: ExtractJwt.fromAuthHeader()
    };

    let strategy = new Strategy(params, async (payload, done) => {

      console.log(payload)

      const valid_user = await connection.findOne({email: payload.email});
      connection.close();
      if (valid_user) {
        return done(null, valid_user);
      } else {
        return done(null, false);
      }
    });
    passport.use(strategy);
    this.app.use(passport.initialize())
  }

  configureRoutes() {

    // login user
    this.app.post('/login', db( async (req, res, next) => {
      const {email, password} = req.body;
      const {Model} = req;
      //search user
      Model.findOne({ email }).then((user) => {
        //compare passwords
        user.comparePassword(password, (err, isMatch) => {
          //server error
          if (err) return res.status(500).send("internal server error");

          if (!isMatch) return res.status(404).send("invalid password");

          //create token
          const token = jwt.sign({ email: user.email }, config.jwtSecret)

          //send token and message
          return res.status(200).send({
            message: "login",
            user: {
              id: user._id,
              name: user.name,
              lastName: user.lastName,
              email: user.email,
              type: user.type,
              status: user.status
            },
            token
          })
        })
      })
    }));

    this.app.post('/register', db(async (req, res, next) => {
      const {email, name, lastName, type, rol} = req.body;
      const {Model} = req;
      try{
        const user = await new Model({name, lastName, email, rol, type}).save();

        res.json(user);

      }catch (e) {
        res.json(e);

      }
    }));

    this.app.post('/update-password/:token', async(req, res) => {
      const {Model} = req;
      const {email, oldPassword, newPassword } = req.body;
      try{
        const user = await Model.findOne({email});
        if(!user) res.status(404).send({message: "user not found"});
        user.comparePassword(oldPassword, (err, isMatch) => {
          if(!isMatch){
            res.status(403).send({message: 'incorrect oldPassword'})
          }else{
            user.password = newPassword;
            user.changePassword = true;
            user.save().then(() => {
              res.status(200).send({message: 'change password ok'})
            })
          }
        })
      }catch(error){
        res.status(500).send({message: "internal server error"})
      }
    })

    this.app.post('/reset-password', async(req, res) => {
      // const {email} = req.body;
      // const {Model} = req;
      //
      // const user = await Model.find({email});
      //
      // if (user){
      //   const token = jwt.sign({email: user.email}, config.secretToken, {expiresIn: 3600});
      //   const url = `http:localhost:4200/reset-password/${token}`;
      //   //this.sendMail.sendEmailUpdatePassword(user.email, `${user.firstName} ${user.lastName}`, res, next);
      // }
      // res.statusCode = 400;
      // res.statusMessage = httpStatus[400];
      // res.end();
      const { email } = req.body;

      const user = await User.findOne({email})

      if(!user) res.status(404).send({message: 'user not found'})

      const randomPassword = user.generateRandomPassword();
      user.changePassword = false;
      user.save().then(() => {
        sendGrid.resetPassword(email, 'noreply@docuflux.com', {
          email,
          randomPassword
        })
        res.status(200).send({message: 'ok'})
      }).catch((err)=>{
        res.status(500).send({message: 'internal error'})
      })
    });
  }
}

exports.SetupAuth = SetupAuth;
