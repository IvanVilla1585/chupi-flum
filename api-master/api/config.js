// @flow
import dotenv from 'dotenv';

dotenv.config({ silent: true });

export default {
  port: process.env.PORT || 3110,
  jwtSecret: process.env.JWTSECRET || 'your secret',
  jwtSession: { session: false },
  db: process.env.MONGO_URL || 'mongodb://localhost/chupiflum',
  userUrl: process.env.USER_URL || 'http://localhost:3000',
  categoryUrl: process.env.CATEGORY_URL || 'http://localhost:3001',
  unitUrl: process.env.UNIT_URL || 'http://localhost:3002',
  presentationUrl: process.env.PRESENTATION_URL || 'http://localhost:3003',
  machineUrl: process.env.MACHINE_URL || 'http://localhost:3004',
  processUrl: process.env.PROCESS_URL || 'http://localhost:3005',
  sengridApiKey: process.env.SENGRID_API_KEY || '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'
};
