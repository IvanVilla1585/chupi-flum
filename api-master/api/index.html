<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>DDOCS API Getway</title>
  </head>
  <body>
    <h2>Auth</h2>
    <ul>
      <li>POST /register</li>
      <li>POST /login</li>
      <li>
        Go to <a href="/graphiql">/graphiql</a>
        to run some GraphQL queries against this server using GraphiQL.
      </li>
    </ul>
  </body>
</html>
