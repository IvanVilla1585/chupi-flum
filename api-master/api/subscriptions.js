// @flow
import { PubSub } from 'graphql-subscriptions';

export const pubsub = new PubSub();

export const CREATED_RADICATED  = 'created_radicated';
export const RADICATED_ADD_DOCUMENT  = 'radicated_add_document';
export const RADICATED_ADD_MEMORANDUM  = 'radicated_add_memorandum';
export const RADICATED_ADD_OFFICE  = 'radicated_add_office';
