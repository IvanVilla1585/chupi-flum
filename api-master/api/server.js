// @flow
import path from 'path';
import express from 'express';
import cors from 'cors';
import { graphqlExpress, graphiqlExpress } from 'graphql-server-express';
import { apolloUploadExpress } from 'apollo-upload-server'
import bodyParser from 'body-parser';
import { createServer } from 'http';

import mongoose from 'mongoose';

import { SubscriptionServer } from 'subscriptions-transport-ws';
import { execute, subscribe } from 'graphql';

// import connectors
import UserConnector from './users/connector';
import CategoryConnector from './categories/connector';
import UnitConnector from './units/connector';
import PresentationsConnector from './presentations/connector';
import MachineConnector from './machines/connector';
import ProcessConnector from './process/connector';

import Email from '../email';
import {SetupAuth} from './auth';

// import setupAuth from './auth';
import schema from './schema';
import config from './config';

const WS_GQL_PATH = '/subscriptions';

export function run(port: string | number) {
  const app = express();

  app.use(cors());
  app.use(bodyParser.json({ limit: '140mb' }));
  app.use(bodyParser.urlencoded({ extended: true, limit: '140mb' }));

  // config auth
  const setupAuth = new SetupAuth(app);


  // set login, register endpoints and JWT configuration
  // const {passport} = setupAuth(app);
  // PROTECT JWT
  //app.use('/graphql', passport.authenticate('jwt', {session: false}), graphqlExpress((req) => {
  app.use('/graphql',
    apolloUploadExpress(),
    graphqlExpress((req) => {
      let user;
      if (req.user) {
        // We get req.user from passport-jwt
        user = req.user;
      }
      return {
        schema,
        context: {
          viewer: user, // logged user
          sendGrid: new Email(config.sengrig_api_key),
          userConnector: new UserConnector(config.userUrl),
          categoryConnector: new CategoryConnector(config.categoryUrl),
          unitConnector: new UnitConnector(config.unitUrl),
          presentationConnector: new PresentationsConnector(config.presentationUrl),
          machineConnector: new MachineConnector(config.machineUrl),
          processConnector: new ProcessConnector(config.processUrl)
        },
      };
    })
  );

  // Serve graphiql (disable for production)
  app.use('/graphiql', graphiqlExpress({
    endpointURL: '/graphql',
    subscriptionsEndpoint: `ws://localhost:${port}${WS_GQL_PATH}`,
    query: 'query loggedUser { me { name, role, createdAt } }',
  }));

  app.get('/', (req: any, res: any) => {
    res.sendFile(path.join(__dirname, 'index.html'));
  });

  const server = createServer(app);

  /**
   * RUN SERVER
   */
  server.listen(port, () => {
    // eslint-disable-next-line no-console
    console.log(`API Server is now running on http://localhost:${port}`);
    // eslint-disable-next-line no-console
    console.log(`API Server over websocket is running on ws://localhost:${port}${WS_GQL_PATH}`);
  });

  /**
   * RUN SERVER SUBSCRIPTION
   */
  new SubscriptionServer({ // eslint-disable-line no-new
    schema,
    execute,
    subscribe,
  }, {
      path: WS_GQL_PATH,
      server,
    });

  return server;
}
