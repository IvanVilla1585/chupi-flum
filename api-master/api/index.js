// @flow
import { run } from './server';
import config from './config';

run(config.port);
