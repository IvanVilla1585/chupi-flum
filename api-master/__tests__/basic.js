import fetch from 'node-fetch';
import { run } from '../api/server';

const testPort = 6789;
const endpointUrl = `http://localhost:${testPort}/graphql`;
const root_endpointUrl = `http://localhost:${testPort}`;

let server;
beforeAll(() => {
  server = run(testPort);
});

it('Server up', async () => {
  const result = await fetch(root_endpointUrl);
  expect(result.status).toBe(200);
});

it('accepts a query', async () => {
  const result = await fetchGraphQL(`
    query loggedUser { me { name, role, createdAt } }
  `);
  const text = await result.text();

  console.log(result.status);
  console.log(text);

  expect(text).toMatchSnapshot();
});


afterAll(() => {
  server.close();
  server = null;
});

function fetchGraphQL(query, variables) {
  return fetch(endpointUrl, {
    method: 'post',
    body: JSON.stringify({ query, variables }),
    headers: { 'content-type': 'application/json' },
  });
}
