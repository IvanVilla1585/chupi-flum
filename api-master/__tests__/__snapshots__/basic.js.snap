// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts a query 1`] = `"{\\"data\\":{\\"me\\":null}}"`;
